### PR TITLE
Decode embedded base64 encoded CQL from within Library resources

### PR DESCRIPTION
**Pull request by plarocque4**
Decode embedded base64 encoded CQL from within Library resources.
Fix issue with R4 Library not having related artifact resources as canonical instead of references.

### CODE REVIEW

**Comment by plarocque4** [comment]
This change goes with the CDS Library and CRD changes to support base64 encoded CQL in the FHIR Library.

File: src/util/fetchArtifacts.js [hunk 1]
```diff
@@ -137,37 +139,62 @@ function fetchArtifacts(fhirPrefix, filePrefix, questionnaireReference, fhirVers
...
+        // base64 decode
+        let elmString = atob(base64elmData);
+
+        // parse the json string
+        let elm = JSON.parse(elmString);
+
+        // set the elm where it needs to be
         if (isMain) {
-          retVal.mainLibraryElms.push(elm);
+          retVal.mainLibraryElms.push(elm)
```
Comment (KeeyanGhoreshi) [inline src/util/fetchArtifacts.js:156]:
```suggestion
          retVal.mainLibraryElms.push(elm);
```

**Review by KeeyanGhoreshi — APPROVED** [review]
Left two comments for very minor issues, otherwise looks good.

File: src/util/fetchArtifacts.js [hunk 2]
```diff
@@ -137,37 +139,62 @@ function fetchArtifacts(fhirPrefix, filePrefix, questionnaireReference, fhirVers
     }
 
     function fetchElmFile(libraryResource, isMain){
-      const elmUri = libraryResource.content.filter(c => c.contentType == "application/elm+json")[0].url;
-      let elmUrl = buildFileUrl(elmUri);
+      if (libraryResource.content[0].url == null) {
```
[the remark this answers is not in the document]
Comment (plarocque4) [reply src/util/fetchArtifacts.js]: Fixed, thanks.

File: src/util/fetchArtifacts.js (hunk 1 above)
Comment (plarocque4) [reply src/util/fetchArtifacts.js]: Fixed, thanks.

**Pull request merged**